qcodes/tests/test_station.py

```diff
@@ -201,3 +215,306 @@ def test_station_after_instrument_is_closed():
+    assert len(mock.parameters) == 2  # there is also IDN
+
+    # special case of `name`
+    mock = st.load_instrument('mock', name='test')
+    assert mock.name == 'test'
+    assert st.components['test'] is mock
+
+    # test address
+    sims_path = get_qcodes_path('instrument', 'sims')
+    st = station_from_config_str(f"""
+instruments:
+  lakeshore:
+    driver: qcodes.instrument_drivers.Lakeshore.Model_336
+    type: Model_336
+    enable_forced_reconnect: true
+    address: GPIB::2::INSTR
+    init:
+      visalib: '{sims_path}lakeshore_model336.yaml@sim'
+    """)
+    st.load_instrument('lakeshore')
+
+
+def test_setup_alias_parameters():
+    st = station_from_config_str("""
+instruments:
+  mock:
+    driver: qcodes.tests.instrument_mocks
+    type: DummyInstrument
+    enable_forced_reconnect: true
+    init:
+      gates: {"ch1"}
+    parameters:
+      ch1:
+        unit: mV
+        label: main gate
+        scale: 2
+        offset: 1
+        limits: -10, 10
+        alias: gate_a
+        initial_value: 9
+
+    """)
+    mock = st.load_instrument('mock')
+    p = getattr(mock, 'gate_a')
+    assert isinstance(p, qcodes.Parameter)
+    assert p.unit == 'mV'
+    assert p.label == 'main gate'
+    assert p.scale == 2
+    assert p.offset == 1
+    assert isinstance(p.vals, validators.Numbers)
+    assert str(p.vals) == '<Numbers -10.0<=v<=10.0>'
+    assert p() == 9
+    mock.ch1(1)
+    assert p() == 1
+    p(3)
+    assert mock.ch1() == 3
+    assert p.raw_value == 7
+    assert mock.ch1.raw_value == 7
+
+def test_setup_delegate_parameters():
+    st = station_from_config_str("""
+instruments:
+  mock:
+    driver: qcodes.tests.instrument_mocks
+    type: DummyInstrument
+    enable_forced_reconnect: true
+    init:
+      gates: {"ch1"}
+    parameters:
+      ch1:
+        unit: V
+        label: ch1
+        scale: 1
+        offset: 0
+        limits: -10, 10
+    add_parameters:
+      gate_a:
+        source: ch1
+        unit: mV
+        label: main gate
+        scale: 2
+        offset: 1
+        limits: -6, 6
+        initial_value: 2
+
+    """)
+    mock = st.load_instrument('mock')
+    p = getattr(mock, 'gate_a')
+    assert isinstance(p, DelegateParameter)
+    assert p.unit == 'mV'
+    assert p.label == 'main gate'
+    assert p.scale == 2
+    assert p.offset == 1
+    assert isinstance(p.vals, validators.Numbers)
+    assert str(p.vals) == '<Numbers -6.0<=v<=6.0>'
+    assert p() == 2
+    assert mock.ch1.unit == 'V'
+    assert mock.ch1.label == 'ch1'
+    assert mock.ch1.scale == 1
+    assert mock.ch1.offset == 0
+    assert isinstance(p.vals, validators.Numbers)
+    assert str(mock.ch1.vals) == '<Numbers -10.0<=v<=10.0>'
+    assert mock.ch1() == 5
+    mock.ch1(7)
+    assert p() == 3
+    assert p.raw_value == 7
+    assert mock.ch1.raw_value == 7
+    assert (json.dumps(mock.ch1.snapshot()) ==
+            json.dumps(p.snapshot()['source_parameter']))
```

Comment:
why comparing JSON dumps? if i remember correctly, pytest should be able to show reasonable introspection on asserting dictionaries.

Reply:
These are not dictionaries but yaml reader objects. I am comparing the json, because that is what eventually will be written to the db.

Reply:
do you mean that `mock.ch1.snapshot()` is a yaml reader object? how did that happen? so far as i understood, `mock` is a valid qcodes Instrument, isn't it?

> I am comparing the json, because that is what eventually will be written to the db.

hmm. but isn't the intention to ensure that the DelegateParameter's source_parameter is "equal" to the `ch1` parameter? and the taken approach is just to compare the snapshots? I can agree with the convenience of comparing the snapshot dicts, but not with comparing snapshot JSON strings, and my reasoning is that in case something is wrong it's much more convenient to let pytest show where the difference is in a dict instead of a cryptic JSON string. Also, i don't think this test really care about the actual format in which the snapshot is stored in some qcodes DB, it just needs to ensure that the "parameter delegation" is correct. Does this make sense?